thirdparty/patchmatchnet/models/patchmatch.py

```diff
@@ -0,0 +1,965 @@
+                ref_feature=ref_feature,
+                src_features=src_features,
+                ref_proj=ref_proj,
+                src_projs=src_projs,
+                depth_sample=depth_sample,
+                depth_min=depth_min,
+                depth_max=depth_max,
+                iter=iter,
+                grid=eval_grid,
+                weight=weight,
+                view_weights=view_weights,
+            )
+            depth_sample = depth_sample.unsqueeze(1)
+            depth_samples.append(depth_sample)
+
+        for iter in range(2, self.patchmatch_iteration + 1):
+            # local perturbation based on previous result
+            depth_sample = self.depth_initialization(
+                False, depth_min, depth_max, height, width, self.patchmatch_interval_scale, device, depth_sample
+            )
+
+            # adaptive propagation
+            if self.propagate_neighbors > 0:
+                # last iteration on stage 1 does not have propagation (photometric consistency filtering)
+                if not (self.stage == 1 and iter == self.patchmatch_iteration):
+                    depth_sample = self.propagation(
+                        batch=batch,
+                        height=height,
+                        width=width,
+                        depth_sample=depth_sample,
+                        grid=propa_grid,
+                        depth_min=depth_min,
+                        depth_max=depth_max,
+                        depth_interval_scale=self.patchmatch_interval_scale,
+                    )
+            # weights for adaptive spatial cost aggregation in adaptive evaluation
+            weight = depth_weight(
+                depth_sample=depth_sample.detach(),
+                depth_min=depth_min,
+                depth_max=depth_max,
+                grid=eval_grid.detach(),
+                patchmatch_interval_scale=self.patchmatch_interval_scale,
+                evaluate_neighbors=self.evaluate_neighbors,
+            )
+            weight = weight * feature_weight.unsqueeze(1)
+            weight = weight / torch.sum(weight, dim=2).unsqueeze(2)
+
+            # evaluation, outputs regressed depth map
+            depth_sample, score = self.evaluation(
+                ref_feature=ref_feature,
+                src_features=src_features,
+                ref_proj=ref_proj,
+                src_projs=src_projs,
+                depth_sample=depth_sample,
+                depth_min=depth_min,
+                depth_max=depth_max,
+                iter=iter,
+                grid=eval_grid,
+                weight=weight,
+                view_weights=view_weights,
+            )
+
+            depth_sample = depth_sample.unsqueeze(1)
+            depth_samples.append(depth_sample)
+
+        return depth_samples, score, view_weights
+
+
+class SimilarityNet(nn.Module):
+    """Similarity Net, used in Evalutaion class (adaptive evaluation step)

```

Comment:
typo here: "used in Evaluation class"

Reply:
I might call it "Evaluation module" instead of class. This conveys the semantics, instead of just implementation